packages/analytics-browser/src/browser-client.ts

```diff
@@ -145,8 +153,30 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
       this.q.push(this.setSessionId.bind(this, sessionId));
       return;
     }
+    const previousSessionId = this.getSessionId();
+    const previousLastEventTime = this.config.lastEventTime;
+
     this.config.sessionId = sessionId;
     this.config.lastEventTime = undefined;
+
+    if (isSessionTrackingEnabled(this.config.defaultTracking)) {
+      if (previousSessionId && previousLastEventTime) {
+        const eventOptions: EventOptions = {
+          session_id: previousSessionId,
+          time: previousLastEventTime + 1,
+        };
+        if (this.previousSessionUserId) {
+          eventOptions.user_id = this.previousSessionUserId;
+        }
+        this.track(END_SESSION_EVENT, undefined, eventOptions);
+        this.previousSessionUserId = undefined;
+      }
+
+      this.track(START_SESSION_EVENT, undefined, {
+        session_id: sessionId,
+        time: sessionId - 1,
```

Comment:
Is there any reason we `-1` here?

Reply:
I added this to guarantee that this is the first session of the event. The times must be manipulated so that analytics sees them in the right order